tests/Rules/NarrowPublicClassMethodParamTypeRule/Fixture/Generics/SkipPassedGenerics.php

```diff
@@ -15,10 +15,10 @@ class User
     public function doFoo(GenericA $g):void
     {
     }
-}
 
-/** @param GenericA<string> $g */
-function doFoo($g):void {
-    $user = new User();
-    $user->doFoo($g);
+    /** @param GenericA<string> $g */
+    function doBar($g):void {
+        $this->doFoo($g);
+    }
```

Comment:
for some reason I don't understand the test-suite only errors for this case on first-class callables, while in real world projects we see the same error on non-first-class-callables